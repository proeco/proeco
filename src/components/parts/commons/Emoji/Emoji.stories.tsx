import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';
import { Emoji } from './Emoji';

export default {
  title: 'parts/commons/atoms/Emoji',
  component: Emoji,
  argTypes: { onClose: { action: 'closed' } },
} as ComponentMeta<typeof Emoji>;

const Template: ComponentStory<typeof Emoji> = (args) => {
  return (
    <Box p="32px" display="flex" gap="16px">
      <Emoji emojiId={args.emojiId} size={24} />
      <Emoji emojiId={args.emojiId} size={32} />
      <Emoji emojiId={args.emojiId} size={40} />
    </Box>
  );
};

export const DirectoryEmoji = Template.bind({});

DirectoryEmoji.args = {
  emojiId: 'open_file_folder',
};
