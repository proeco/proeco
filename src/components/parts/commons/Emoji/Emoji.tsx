import { VFC } from 'react';
import { EmojiData, Emoji as EmojiOrigin } from 'emoji-mart';
import { Box, styled } from '@mui/system';

type Props = {
  emojiId: EmojiData | string;
  size: number;
  onClick?: () => void;
};

export const Emoji: VFC<Props> = ({ emojiId, size, onClick }) => {
  return (
    <StyledEmojiWrapper onClick={onClick}>
      <EmojiOrigin emoji={emojiId} size={size} />
    </StyledEmojiWrapper>
  );
};

const StyledEmojiWrapper = styled(Box)`
  .emoji-mart-emoji {
    display: block;
  }
`;
