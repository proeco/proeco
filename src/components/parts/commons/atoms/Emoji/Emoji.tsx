import { VFC } from 'react';
import { EmojiData, Emoji as EmojiOrigin } from 'emoji-mart';
import { Box, styled } from '@mui/system';

type Size = 'sm' | 'md' | 'lg';

type Props = {
  emojiId: EmojiData | string;
  size?: Size;
  onClick?: () => void;
};

const sizeMap: { [key in Size]: number } = {
  sm: 24,
  md: 28,
  lg: 32,
};

export const Emoji: VFC<Props> = ({ emojiId, size = 'md', onClick }) => {
  return (
    <StyledEmojiWrapper onClick={onClick}>
      <EmojiOrigin emoji={emojiId} size={sizeMap[size]} />
    </StyledEmojiWrapper>
  );
};

const StyledEmojiWrapper = styled(Box)`
  .emoji-mart-emoji {
    display: block;
  }
`;
