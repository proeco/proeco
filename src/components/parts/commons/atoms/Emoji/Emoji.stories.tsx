import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';
import { Emoji } from './Emoji';

export default {
  title: 'parts/commons/atoms/Emoji',
  component: Emoji,
  argTypes: { onClose: { action: 'closed' } },
} as ComponentMeta<typeof Emoji>;

const Template: ComponentStory<typeof Emoji> = (args) => {
  return (
    <Box p="32px" display="flex" gap="16px">
      <Emoji emojiId={args.emojiId} size="sm" />
      <Emoji emojiId={args.emojiId} size="md" />
      <Emoji emojiId={args.emojiId} size="lg" />
    </Box>
  );
};

export const DirectoryEmoji = Template.bind({});

DirectoryEmoji.args = {
  emojiId: 'open_file_folder',
};
