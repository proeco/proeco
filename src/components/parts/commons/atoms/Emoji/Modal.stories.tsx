import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Box } from '@mui/system';
import { Emoji } from './Emoji';

export default {
  title: 'parts/commons/atoms/Emoji',
  component: Emoji,
  argTypes: { onClose: { action: 'closed' } },
} as ComponentMeta<typeof Emoji>;

const Template: ComponentStory<typeof Emoji> = (args) => {
  return (
    <Box p="32px" display="flex" gap="16px">
      <Emoji emojiId={args.emojiId} size="small" />
      <Emoji emojiId={args.emojiId} size="medium" />
      <Emoji emojiId={args.emojiId} size="large" />
    </Box>
  );
};

export const DirectoryEmoji = Template.bind({});

DirectoryEmoji.args = {
  emojiId: 'open_file_folder',
};
