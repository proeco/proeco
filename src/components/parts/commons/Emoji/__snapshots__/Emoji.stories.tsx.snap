// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots parts/commons/Emoji Directory Emoji 1`] = `
<div
  className="MuiBox-root css-4e1wul"
>
  <div
    className="MuiBox-root css-186alq0"
  >
    <span
      aria-label="ðŸ“‚, open_file_folder"
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png)",
            "backgroundPosition": "46.42857142857143% 42.85714285714286%",
            "backgroundSize": "5700% 5700%",
            "display": "inline-block",
            "height": 24,
            "width": 24,
          }
        }
      />
    </span>
  </div>
  <div
    className="MuiBox-root css-186alq0"
  >
    <span
      aria-label="ðŸ“‚, open_file_folder"
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png)",
            "backgroundPosition": "46.42857142857143% 42.85714285714286%",
            "backgroundSize": "5700% 5700%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </div>
  <div
    className="MuiBox-root css-186alq0"
  >
    <span
      aria-label="ðŸ“‚, open_file_folder"
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-apple@5.0.1/img/apple/sheets-256/64.png)",
            "backgroundPosition": "46.42857142857143% 42.85714285714286%",
            "backgroundSize": "5700% 5700%",
            "display": "inline-block",
            "height": 40,
            "width": 40,
          }
        }
      />
    </span>
  </div>
</div>
`;
